core/src/main/java/hudson/model/TreeView.java

```diff
@@ -140,9 +140,16 @@ public void deleteView(View view) throws IOException {
     }
 
     public View getView(String name) {
-        for (View v : views)
-            if(v.getViewName().equals(name))
+        for (View v : views) {
+            if (v.getViewName().equals(name))
                 return v;
+            if (v instanceof ViewGroup) {
```

Comment:
Regarding `recursive` I can add it, however, in most cases recursive search should be a desired behavior. Nevertheless, I don't know Jenkins internals and user use cases, so I'm not against that change and can implement it.